cmake/TupleTreeGenerator.cmake

```diff
@@ -229,6 +259,54 @@ function(
     DEPENDS "${YAML_DEFINITIONS}" ${TUPLE_TREE_GENERATOR_SOURCES})
 endfunction()
 
+# Generates typescript files
+function(
+  tuple_tree_generator_generate_typescript
+  # Path to the yaml definitions
+  YAML_DEFINITIONS
+  # Base namespace of the generated classes (e.g. model)
+  NAMESPACE
+  # Type to use as the root of the schema
+  ROOT_TYPE
+  # Name of the global type, e.g. Model
+  GLOBAL_NAME
+  # Files to be included in the prouduced output
+  INCLUDE_FILES
+  # Types equivalent to plain strings
+  STRING_TYPES
+  # Types equivalent to plain strings that get a separate type definition
+  EXTERNAL_TYPES
+  # Output path
+  OUTPUT_PATH)
+
+  set(INCLUDE_FILES_ARGS)
+  foreach(IF ${INCLUDE_FILES})
+    list(APPEND INCLUDE_FILE_ARGS --external-file "${IF}")
+  endforeach()
+
+  set(STRING_TYPE_ARGS)
+  foreach(ST ${STRING_TYPES})
+    list(APPEND STRING_TYPE_ARGS --string-type "${ST}")
+  endforeach()
+
+  set(EXTERNAL_TYPE_ARGS)
+  foreach(ET ${EXTERNAL_TYPES})
+    list(APPEND EXTERNAL_TYPE_ARGS --external-type "${ET}")
+  endforeach()
+
+  add_custom_command(
```

Comment:
I see an `add_custom_command` without an `add_custom_target`.
How does the Python generator do this?
Who injects the dependency on the output of this command?

Reply:
https://github.com/revng/revng/blob/develop/cmake/TupleTreeGenerator.cmake#L85
The output of the command (e.g. `model.ts`) is added as a file-level dependency to `EXTRA_TARGETS` and then all of them are added as targets

Reply:
Uhm OK